crates/sui-e2e-tests/tests/rpc/data/burnonly_coin/sources/burnonly_coin.move

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// Example coin using the BurnOnly supply state in CoinRegistry
+module burnonly_coin::burnonly_coin {
```

Comment:
Quick note re all examples. 
We define modules as:

```move
module burnonly_coin::burnonly_coin;

use ....
```

And we don't use `public entry` - compiler actually lints against it. Just `public` is enough.